contracts/SpokePool.sol

```diff
@@ -548,59 +548,101 @@ abstract contract SpokePool is
+     * use this function to front-run another depositor's unsafe deposit. This function guarantees that the resultant
+     * deposit nonce will not collide with a safe uint256 deposit nonce whose 24 most significant bytes are always 0.
+     * @param depositor See identically named parameter in depositV3() comments.
+     * @param recipient See identically named parameter in depositV3() comments.
+     * @param inputToken See identically named parameter in depositV3() comments.
+     * @param outputToken See identically named parameter in depositV3() comments.
+     * @param inputAmount See identically named parameter in depositV3() comments.
+     * @param outputAmount See identically named parameter in depositV3() comments.
+     * @param destinationChainId See identically named parameter in depositV3() comments.
+     * @param exclusiveRelayer See identically named parameter in depositV3() comments.
+     * @param quoteTimestamp See identically named parameter in depositV3() comments.
+     * @param fillDeadline See identically named parameter in depositV3() comments.
+     * @param exclusivityPeriod See identically named parameter in depositV3() comments.
+     * @param message See identically named parameter in depositV3() comments.
+     */
+    function unsafeDepositV3(
+        address depositor,
+        address recipient,
+        address inputToken,
+        address outputToken,
+        uint256 inputAmount,
+        uint256 outputAmount,
+        uint256 destinationChainId,
+        address exclusiveRelayer,
+        uint96 depositNonce,
+        uint32 quoteTimestamp,
+        uint32 fillDeadline,
+        uint32 exclusivityPeriod,
+        bytes calldata message
+    ) public payable nonReentrant unpausedDeposits {
+        // @dev Create the uint256 deposit ID by concatenating the address (which is 20 bytes) with the 12 byte
+        // depositNonce parameter. The resultant 32 byte string can be casted to an "unsafe" uint256 deposit ID.
+        // This guarantees the resultant ID won't collide with a "safe" deposit ID which is set by
+        // implicitly casting a uint32 to a uint256, where the left-most 24 bytes are set to 0. We guarantee
+        // that there are no collisions between "unsafe" and "safe" deposit ID's as long as the msg.sender's address
+        // is not the zero address.
+
+        // @dev For some L2's, it could be possible that the zero address can be the msg.sender which might
+        // make it possible for an unsafe deposit hash to collide with a safe deposit nonce. To prevent these cases,
+        // we might want to consider explicitly checking that the msg.sender is not the zero address. However,
+        // this is unlikely and we choose to not check it and incur the extra gas cost:
+        // e.g. if (msg.sender == address(0)) revert InvalidUnsafeDepositor();
+
+        uint256 depositId = uint256(bytes32(abi.encodePacked(msg.sender, depositNonce)));
```

Comment:
We could flip the ordering and require that depositNonce > 0, making every custom nonce a very large number regardless of address. That would mean that any address would work, which would remove any concerns around precompiles that are at address 0x1, 0x2, 0x3 (not sure if anybody does this, but would prefer to avoid building our noncing around it).

Thoughts?

Reply:
> require that depositNonce > 0,

Ideally I was trying to avoid an additional `require` statement here but I agree this would be a good algorithm. This is why I wanted to avoid `require(msg.sender != address(0))`

Reply:
I think this comment is irrelevant now that we hash the packed data: 19a33801e31da303616ee12e192589941528eb1c